### PR TITLE
implemented process-name

### LINKED ISSUE #331: Port process-name to Rust
```c
DEFUN ("process-name", Fprocess_name, Sprocess_name, 1, 1, 0,
       doc: /* Return the name of PROCESS, as a string.
This is the name of the program invoked in PROCESS,
possibly modified to make it unique among process names.  */)
  (register Lisp_Object process)
{
  CHECK_PROCESS (process);
  return XPROCESS (process)->name;
}
```

### PR DESCRIPTION
**Pull request by Yarikx**
Closes #331

### CODE REVIEW

File: rust_src/remacs-sys/lib.rs [hunk 1]
```diff
@@ -736,6 +736,8 @@ pub struct Lisp_Char_Table {
 
 #[repr(C)]
 pub struct Lisp_Process {
+    pub header: Lisp_Vectorlike_Header,
```
Comment (birkenfeld) [inline rust_src/remacs-sys/lib.rs]: So... nobody was using the members before?

File: rust_src/src/process.rs [hunk 2]
```diff
@@ -7,10 +7,17 @@ use lists::{assoc, cdr};
...
+    fn name(&self) -> LispObject {
+        return LispObject::from_raw(self.name);
+    }
+}
+
 /// Return t if OBJECT is a process.
 #[lisp_fn]
 pub fn processp(object: LispObject) -> LispObject {
-    LispObject::from_bool(object.is_process())
+    return LispObject::from_bool(object.is_process());
```
Comment (birkenfeld) [inline rust_src/src/process.rs]: unnecessary (and unidiomatic) change, please revert

File: rust_src/src/process.rs [hunk 3]
```diff
@@ -7,10 +7,17 @@ use lists::{assoc, cdr};
 
 pub type LispProcessRef = ExternalPtr<Lisp_Process>;
 
+impl LispProcessRef {
+    #[inline]
+    fn name(&self) -> LispObject {
+        return LispObject::from_raw(self.name);
```
Comment (birkenfeld) [inline rust_src/src/process.rs]: `return` can be removed here too

File: rust_src/remacs-sys/lib.rs (hunk 1 above)
Comment (Yarikx) [reply rust_src/remacs-sys/lib.rs]: Apparently. I was getting `tty_name` instead of `name` there.

File: rust_src/src/process.rs (hunk 2 above)
Comment (Yarikx) [reply rust_src/src/process.rs]: Added it because of failing build on Travis. I don't like that return either.

**Pull request merged**